include/NProposingTiesMatching.h

```diff
@@ -0,0 +1,27 @@
+#ifndef N_PROPOSING_TIES_MATCHING_H
+#define N_PROPOSING_TIES_MATCHING_H
+
+#include "MatchingAlgorithm.h"
+#include "PreferenceList.h"
+#include "VertexBookkeeping.h"
+#include <stack>
+#include <ostream>
+
+// simulate Gale and Shapley with n levels for proposing side when ties are present in the preference lists in order to get a weak stable matching
```

Comment:
Capitalize first letter of the sentence, end it with a full stop, and make the line length 80 characters.